src/game_api/layer.cpp

```diff
@@ -215,3 +215,35 @@ void Layer::move_grid_entity(Entity* ent, uint32_t x, uint32_t y, Layer* dest_la
         }
     }
 }
+
+void Layer::destroy_grid_entity(Entity* ent)
+{
+    if (ent)
+    {
+        auto items = ent->items.entities();
+        for (auto ptr = items.cend(); ptr != items.cbegin();)
+        {
+            ptr--;
+            Entity* item_ent = *ptr;
+            if (item_ent->type->search_flags & ~1) // if not player

```

Comment:
this feels very weirdly specific

Reply:
Destroying the player would just crash the game. Although ent->is_player() is a better check for this, like used in UI::safe_destroy

Reply:
I didn't mean this as: why check player, but like: why only check player?

Reply:
As a safeguard, cause it's the only thing that crashes the game. Also I think destroying the items is usually the right thing to do, like with chainbugs on a chain on the chain ceiling you're distroying, but crashing is definitely never what you want.

Reply:
This is pretty theoretical, cause it's a level gen function and would ideally be used before monsters spawn or even chains are grown, but we can still prepare for that.